addon/templates/components/frost-list-item-content.hbs

```diff
@@ -0,0 +1,67 @@
+<div class={{concat
+    'frost-list-item-container'
+    (if (eq index 0) ' first')
+    (if (eq index (sub _items.length 1)) ' last')
+    (if model.states.isSelected ' is-selected')
+  }}
+  data-test={{hook (concat hook '-item-container') index=index }}
+>
+  <div class='frost-list-item-container-base'>
+    {{#if (and (or itemExpansion isTypedItemWithExpansion) (not isTypedItemWithoutExpansion))}}
+      {{frost-list-item-expansion
+        hook=(concat hookPrefix '-expansion')
+        hookQualifiers=(hash index=index)
+        model=model.content
+        isExpanded=model.states.isExpanded
+        onExpand=onExpand
+      }}
+    {{else if isAnyTypedItemExpansion}}
+      <div class="item-expansion-placeholder"></div>
+    {{/if}}
+
+    {{#if onSelectionChange}}
+      {{frost-list-item-selection
+        hook=(concat hookPrefix '-selection')
+        hookQualifiers=(hash index=index)
+        model=model.content
+        isSelected=model.states.isSelected
+        onSelect=onSelect
+      }}
+    {{/if}}
+
+    {{component itemComponent
+      hook=(concat hookPrefix '-item')
+      hookQualifiers=(hash index=index)
+      model=model.content
+      isExpanded=model.states.isExpanded
+      isSelected=model.states.isSelected
+      onSelect=onSelect
+    }}
+  </div>
+
+  {{#if model.states.isExpanded}}
+    <div class='frost-list-item-container-expansion'>
+      {{#if isTypedItemWithExpansion}}
+        {{component typedItemExpansionComponent
+          hook=(concat hookPrefix '-itemExpansion')
+          hookQualifiers=(hash index=index)
+          model=model.content
+          isExpanded=model.states.isExpanded
+          isSelected=model.states.isSelected
+          onSelect=onSelect
+        }}
+      {{else if isTypedItemWithoutExpansion}}
```

Comment:
Why? 

The way I'm seeing this in the template you should only do something like:
```
{{component _itemExpansion
hook=(concat hookPrefix '-itemExpansion')
hookQualifiers=(hash index=index)
model=model.content
isExpanded=model.states.isExpanded
isSelected=model.states.isSelected
onSelect=onSelect
}}
```

where `_itemExpansion` is type or not.

Reply:
Yes that's exactly it. If the item is typed and has expansion, then we want to show the typed item expansion. If the item is typed, but has no expansion, then we don't want to fall back to default expansion. If the item is not typed, then we want to show the default expansion. The reason why I haven't made a CP that handles all of this is because I cannot send the result of that to the component helper because the value passed to the component helper cannot be empty or undefined.